### PR TITLE
[API] Organizer can create an event

### PR DESCRIPTION
**Pull request by shahin1290**
## Description
```
As an Organizer
In order to engage my community
I would like to be able to create events for my group members
```

## PivotalTracker
[PT Link](https://www.pivotaltracker.com/story/show/163513750)

## Tasks
- Setup Event#create action + add an endpoint
- Create request specs

### CODE REVIEW

File: spec/requests/organizer_can_create_event_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+describe 'POST /events' do
+  let(:user) { create(:user) }
+  let!(:group_1) { create(:group) } 
+
```
Comment (CamronLDNF) [inline spec/requests/organizer_can_create_event_spec.rb:4]: Minor thing but no need to call it ":group_1" as there is only one group being created for the test. Rename to ":group"

**Review by CamronLDNF — COMMENTED** [review]
See (one) comment below.

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -7,4 +7,19 @@ def index
     end
     render json: events, each_serializer: Events::IndexSerializer
   end
+
+  def create
+    event = Event.new(event_params.merge(group_id: params[:group_id]))
```
Comment (tochman) [inline app/controllers/events_controller.rb]:
```
group = Group.find(params[:group_id])
event = group.events.create(event_params)
if event.persisted?
   render.....
```

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -7,4 +7,19 @@ def index
...
+    if event.save
+      
+      render json: { message: 'You have created an event successfully' }
+    else
+      render json: { error: event.errors.full_messages }, status: 400
```
Comment (tochman) [inline app/controllers/events_controller.rb]: You can omit ` status: 400`

File: spec/requests/organizer_can_create_event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(user_credentials) }
+    
+    before do
+      post "/groups/#{group.id}/events", params: { event: {title: 'Craft Academy'} }, headers: headers
+    end
+
+    it 'responds with success message' do
+      expect(response_json['message']).to eq 'You have created an event successfully'
+    end
+
```
Comment (CamronLDNF) [inline spec/requests/organizer_can_create_event_spec.rb:20]: Rephrase to 'You have successfully created an event'

File: app/controllers/events_controller.rb (hunk 2 above)
Comment (Jon1969Edwards) [reply app/controllers/events_controller.rb]: Done

File: app/controllers/events_controller.rb (hunk 3 above)
Comment (Jon1969Edwards) [reply app/controllers/events_controller.rb]: Done

File: spec/requests/organizer_can_create_event_spec.rb (hunk 4 above)
Comment (Jon1969Edwards) [reply spec/requests/organizer_can_create_event_spec.rb:20]: Done

**Pull request merged**